x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -159,6 +158,8 @@ public ClusterState execute(final ClusterState currentState) throws Exception {
                 Map<String, String> metadata = new HashMap<>();
                 metadata.put(Ccr.CCR_CUSTOM_METADATA_LEADER_INDEX_SHARD_HISTORY_UUIDS, String.join(",", historyUUIDs));
                 metadata.put(Ccr.CCR_CUSTOM_METADATA_LEADER_INDEX_UUID_KEY, leaderIndexMetaData.getIndexUUID());
+                metadata.put(Ccr.CCR_CUSTOM_METADATA_LEADER_INDEX_NAME_KEY, leaderIndexMetaData.getIndex().getName());
+                metadata.put(Ccr.CCR_CUSTOM_METADATA_LEADER_CLUSTER_NAME_KEY, request.getLeaderCluster());
```

Comment:
By remember this two additional key value pairs the resume api no longer needs to accept a leader index / cluster parameters.

The leader index name isn't really necessary as the leader index uuid can also be used to lookup the leader index metadata in the leader cluster. The `CcrLicenseChecker#checkRemoteClusterLicenseAndFetchLeaderIndexMetadataAndHistoryUUIDs(...)` only accepts a leader index name now. Put follow api also uses this method and that would mean I another version of this method is needed to knows how to work with index uuid. I found that this would make `CcrLicenseChecker` too complicated, so I decided to remember the leader index name inside ccr custom index metadata instead.